index.js

```diff
@@ -405,7 +405,22 @@ SendStream.prototype.onStatError = function onStatError (error) {
  */
 
 SendStream.prototype.isFresh = function isFresh () {
-  return fresh(this.req.headers, this.res._headers)
+  var resHeaders = this.res._headers || {}
+
+  // Support changes in node with how response headers are stored
+  var headerNames = this.res._headerNames || {}
+  var keys = Object.keys(resHeaders)
+  if (keys.length > 0 && headerNames[keys[0]] === undefined) {
```

Comment:
I think I understand what this check is attempting to do, but the popular https://github.com/molnarg/node-http2 module does not emulate the `_headerNames` property, which I assume this check then would assume that `_headers` object is the new style rather than the old style, is that correct?